scrapyd-client/scrapyd-deploy

```diff
@@ -52,24 +52,35 @@ def parse_opts():
         help="use the given egg, instead of building it")
     parser.add_option("--build-egg", metavar="FILE",
         help="only build the egg, don't deploy it")
+    parser.add_option("--settings-module", metavar="SCRAPY_SETTINGS_MODULE",
+        help="python module to load as scrapy settings, can be passed also "
+        "through the env var SCRAPY_SETTINGS_MODULE")
+    parser.add_option("--url", metavar="SCRAPYD_URL",
+        help="URL to the scrapyd server, overrides the url definition in the "
+        "target if any")
     return parser.parse_args()
 
 def main():
     opts, args = parse_opts()
     exitcode = 0
+    if opts.settings_module:
+        os.environ['SCRAPY_SETTINGS_MODULE'] = opts.settings_module
+
     if not inside_project():
-        _log("Error: no Scrapy project found in this location")
+        _log("Error: no Scrapy project found in this location and no url"
```

Comment:
Because if you pass the url, there's no need to have a scrapy.cfg file, so you don't need to be inside a `scrapy` project, just a regular python module will do. I can rephrase if you have a better idea.